go-controller/pkg/ovn/egressgw.go

```diff
@@ -121,46 +245,24 @@ func (oc *Controller) addGWRoutesForNamespace(namespace string, egress gatewayIn
 				oc.deletePerPodGRSNAT(pod.Spec.NodeName, portInfo.ips)
 			}
 		}
-		gr := util.GetGatewayRouterFromNode(pod.Spec.NodeName)
-		prefix, err := oc.extSwitchPrefix(pod.Spec.NodeName)
-		if err != nil {
-			klog.Infof("Failed to find ext switch prefix for %s %v", pod.Spec.NodeName, err)
-			continue
-		}
-
-		port := prefix + types.GWRouterToExtSwitchPrefix + gr
-		for _, gw := range egress.gws {
-			for _, podIP := range pod.Status.PodIPs {
-				if utilnet.IsIPv6(gw) != utilnet.IsIPv6String(podIP.IP) {
-					continue
-				}
-
-				// if route was already programmed, skip it
-				if foundGR, ok := nsInfo.podExternalRoutes[podIP.IP][gw.String()]; ok && foundGR == gr {
-					continue
-				}
-
-				mask := GetIPFullMask(podIP.IP)
-				nbctlArgs := []string{"--may-exist", "--policy=src-ip", "--ecmp-symmetric-reply",
-					"lr-route-add", gr, podIP.IP + mask, gw.String(), port}
-				if egress.bfdEnabled {
-					nbctlArgs = []string{"--may-exist", "--bfd", "--policy=src-ip", "--ecmp-symmetric-reply",
-						"lr-route-add", gr, podIP.IP + mask, gw.String(), port}
-				}
-
-				_, stderr, err := util.RunOVNNbctl(nbctlArgs...)
 
-				if err != nil && !strings.Contains(stderr, DuplicateECMPError) {
-					return fmt.Errorf("unable to add src-ip route to GR router, stderr:%q, err:%v", stderr, err)
-				}
-				if err := oc.addHybridRoutePolicyForPod(net.ParseIP(podIP.IP), pod.Spec.NodeName); err != nil {
-					return err
-				}
-				if nsInfo.podExternalRoutes[podIP.IP] == nil {
-					nsInfo.podExternalRoutes[podIP.IP] = make(map[string]string)
-				}
-				nsInfo.podExternalRoutes[podIP.IP][gw.String()] = gr
+		podIPs := make([]*net.IPNet, 0)
+		for _, podIP := range pod.Status.PodIPs {
+			var fullMask int
+			if utilnet.IsIPv6String(podIP.IP) {
+				fullMask = 128
+			} else {
+				fullMask = 32
 			}
+			cidr := fmt.Sprintf("%s/%d", podIP.IP, fullMask)
+			_, ipNet, err := net.ParseCIDR(cidr)
```

Comment:
```suggestion
			cidr := fmt.Sprintf("%s%s", podIP.IP, GetIPFullMask(podIP.IP))
			_, ipNet, err := net.ParseCIDR(cidr)
```
we could continue using  `GetIPFullMask` to avoid code duplication and keeping this function neat.

Reply:
done